lib/travis/build/addons/apt.rb

```diff
@@ -206,7 +210,10 @@ def add_apt_packages
                 stop_postgresql
               end
 
-              sh.cmd 'travis_apt_get_update', retry: true, echo: true, timing: true
+              unless config[:update]
+                # this is run elsewhere when sources are added
+                sh.cmd 'travis_apt_get_update', retry: true, echo: true, timing: true
+              end
```

Comment:
This will make `update:` work backwards if `packages:` is specified and `sources:` is not.